content/api-streamer/producer.textile

```diff
@@ -0,0 +1,239 @@
+---
+title: Producer
+section: api-streamer
+index: 67
+languages:
+  - javascript
+  - nodejs
+  - php
+  - python
+  - ruby
+  - java
+  - swift
+  - objc
+  - csharp
+  - go
+---
+
+The "Ably API Streamer":https://ably.io/api-streamer makes it easy and cost-effective for organizations and developers to offer realtime APIs, at any scale. It provides a unified solution for publishing, managing and distributing realtime data via open protocol APIs, all built on the "global Ably Data Stream Network":https://www.ably.io/dsn. API Streamer, amongst other things, is being used to share live sports scores and financial market data, mobility IoT data such as GPS locations, and SaaS integration updates in real time.
+
+h2(#publishing-data-streams). Using the API Streamer to publish data streams
+
+You can share your realtime data streams with consumers in three distinct steps: 1) Deploy,  2) Manage and 3) Distribute.
+
+<a href="/images/diagrams/api-streamer-producer-steps.png" target="_blank">
+  <img src="/images/diagrams/api-streamer-producer-steps.png" style="width: 100%" alt="Steps involved in publishing data streams">
+</a>
+
+h3(#deploy). Deploy
+
+Event-driven and realtime data, by it's very nature, needs to flow from the producer to the consumer with the lowest possible latency. Given producers in most cases are decoupled from consumers of data, and not all data that is published is needed by all subscribers, the "publish/subscribe pattern":https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern is most commonly used in systems designed to deliver realtime data as streams of events or messages.
+
+In the first step of publishing data streams, you'll need to set up an app on the Ably app dashboard or use one of the existing apps. By default, all the Ably accounts come with two pre-loaded apps called 'Production' and 'Sandbox'. If you are just trying out the API Streamer for the first time, we recommend you use the 'Sandbox' app to get up and running quickly.
+
+Ably fundamentally organizes all the data into units called "channels":/core-features/channels. In order to deploy your data into your Ably apps, you'll need to publish it to one or more channels.
+
+You can do this by either by using our "Realtime Client SDKs":/realtime, our "REST Client SDKs":/rest, our "MQTT adapter":/mqtt, or directly using our "raw REST API":/rest-api (although this last option isn't recommended as it is inferior to our SDKs in terms of performance and resilience).
+
+In most cases, if you are only publishing your data streams and not expecting a two-way communication between you and the consumer of the data, the most efficient way is to use the REST Client SDKs.
```

Comment:
`not expecting a two-way communication between you and the consumer of the data` don't think using streamer it's possible to have two-way communication.

Reply:
What if they subscribe to their own products? @tomczoink

Reply:
Two-way communication to me implies that not only would they be publishing to a consumer, but the consumer would also be publishing to them. If you want to indicate it's possible they may want to subscribe as well that's fine, but I don't think two-way communication is the right way to describe that.